### DISCUSSION TITLE
How Do I Run RefindPlus When Not Recognised by the Firmware

### DISCUSSION THREAD

**Discussion by btwise** [dakanji/RefindPlus#102] [General]
### Suggested Feature

Some OEM motherboards, such as Lenovo, cannot recognize the refindplus bootloader that has been changed to /EFI/BOOT/BOOTX64.EFI, it can only scan the Microsoft EFI boot file in the ESP partition, and there is no EFI SHELL environment, how to start refindplus .efi?

**Comment by dakanji** [comment]
Ok. Looked again and it seems it is a generic question about how to load the RefindPlus efi on motherboards that have some limiations in what they can load.

Well, that issue is outside RefindPlus' control. I suppose you could put the RefindPlus items in the scanned location and rename the Refindplus efi to match whatever the firmware scans.

Otherwise, check how other bootloaders are used with such units and apply the same to RefindPlus

**Reply by btwise** [reply]
This is not the case. Even if I rename Microsoft's BOOTX64.EFI to other XXX.EFI, it can be scanned in the UEFI BIOS boo menu, but even if refindplus.efi is renamed to BOOTX64.EFI or bootmrgfw.efi, it will not appear in the UEFI boot menu of the motherboard. It can't find this boot item unless I force it into NVRAM with tools like BOOTICE, but this is not convenient. I will use it on many customers' machines, not just a single computer.

**Reply by dakanji** [reply]
So how do you load rEFInd or any other bootloader on the machine? Are you saying Whatever works for those does not work for RefindPlus? That this is something that only affects RefindPlus?

**Reply by Kylejustknows** [reply]
My 2 cents:
Since this is some sort of Microsoft's dirty job, I would say they enforced the "Default Boot manager" to be: 1, inside the EFI/Microsoft path, 2, Must be MS Secure boot signed, regardless of your bios settings.
The rEFInd is not Microsoft signed, so it won't show up.

And for the manually injected 3rd-party.efi, they allow it to be "secure boot off", so Linux users won't turn too mad.

**Comment by startergo** [comment]
For Lenovo Secure boot has to be disabled in the BIOS to boot "unauthorized operating systems". You can also add certificates to the database after signing refind or RP bootloader.
![image](https://user-images.githubusercontent.com/7897244/163796154-83dc83ee-cb56-43da-ad71-90ad7677cd2a.png)
![image](https://user-images.githubusercontent.com/7897244/163796198-c1e4fea8-2970-47c2-9516-842aeb10b8f2.png)

**Comment by startergo** [comment]
This is booting [rEFInd CD image](https://sourceforge.net/projects/refind/files/0.13.3.1/refind-cd-0.13.3.1.zip/download) with [Ventoy ](https://github.com/ventoy/Ventoy) USB formatted drive on my Thinkpad and Secure boot disabled:
![image](https://user-images.githubusercontent.com/7897244/163806315-a62aed69-c520-486a-a15d-ab62e7b3ed9d.png)
